theseus/utils/examples/motion_planning/motion_planner.py

```diff
@@ -20,25 +20,27 @@ def __init__(
         collision_weight: float,
         Qc_inv: List[List[int]],
         num_time_steps: int,
-        optim_method: str,
-        max_optim_iters: int,
-        step_size: float = 1.0,
         use_single_collision_weight: bool = True,
-        device: str = "cpu",
         dtype: torch.dtype = torch.double,
     ):
+        for v in [
+            map_size,
+            epsilon_dist,
+            total_time,
+            collision_weight,
+            Qc_inv,
+            num_time_steps,
+        ]:
+            assert v is not None
+
+        super().__init__(dtype=dtype)
```

Comment:
Did we miss `device`?

Reply:
We don't have a kwarg `device=` in `th.Objective` (maybe we should add later?). In the `MotionPlanner` the objective is pushed to the given device after the objective is created. The code does assume that all inputs are on the same device.

Reply:
Okay. Hmm might be worth having a device in Objective, later.

Reply:
Created #281